server/src/com/treode/server/Resource.scala

```diff
@@ -97,7 +97,30 @@ class Resource (host: HostId, store: SchematicStore) extends Service [Request, R
         respond (req, Status.PreconditionFailed)
     }}
 
+  def batch_write (req: Request): Async [Response] = {
+    val tx = req.transactionId (host)
+    val ct = req.conditionTxClock (TxClock.now)
+    try {
+      store.batch_write(tx, ct, req)
+      .map [Response] { vt =>
+        val rsp = req.response
+        rsp.status = Status.Ok
+        rsp.headerMap.add ("Value-TxClock", vt.toString)
+        rsp
+      }
+      .recover {
+        case _: StaleException => {
+          respond (req, Status.PreconditionFailed)
+        }
+      }
+    } catch {
+      case v: Throwable => {
```

Comment:
I have that to catch the null pointer exceptions that can result in cases where the json objects are ill-formed (Like "tabe":"table1", instead of "table":"table1"). So, can I replace it with 
case v: NullPointerException => throw new BadRequestException ("Message") ? For other exceptions inside the function, I am directly throwing a BadRequestException now. Or is it better if I manually check for each Null pointer exception inside the function and throw the error from there?